compiler/rustc_data_structures/src/sync.rs

```diff
@@ -30,6 +31,35 @@ pub use vec::AppendOnlyVec;
 
 mod vec;
 
+mod mode {
+    use super::Ordering;
+    use std::sync::atomic::AtomicU8;
+
+    const UNINITIALIZED: u8 = 0;
+    const SINGLE_THREAD: u8 = 1;
+    const MULTIPLE_THREADS: u8 = 2;
+
+    static PARALLEL: AtomicU8 = AtomicU8::new(UNINITIALIZED);
+
+    #[inline]
+    pub fn active() -> bool {
+        match PARALLEL.load(Ordering::Relaxed) {
+            SINGLE_THREAD => false,
+            MULTIPLE_THREADS => true,
+            _ => panic!("uninitialized parallel mode!"),
+        }
+    }
+
+    // Only set by the `-Z threads` compile option
+    pub fn set_parallel_mode(parallel: bool) {
```

Comment:
Naming this `sync::mode::set` should be sufficient.

Reply:
Done